lib/jormungandr/test/bench/Latency.hs

```diff
@@ -101,30 +111,106 @@ import Test.Integration.Framework.DSL
     , listTxEp
     , listWalletsEp
     , networkInfoEp
+    , postTxEp
     , postTxFeeEp
     , request
+    , verify
     )
 
 import qualified Cardano.BM.Configuration.Model as CM
 import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
 
 main :: forall t n. (t ~ Jormungandr, n ~ 'Testnet) => IO ()
 main = do
     tvar <- newTVarIO []
     logging <- setupLatencyLogging tvar
 
-    let meanAvg :: [NominalDiffTime] -> Double
-        meanAvg ts = sum (map realToFrac ts) * 1000 / fromIntegral (length ts)
-    let buildResult :: [NominalDiffTime] -> Builder
-        buildResult [] = "ERR"
-        buildResult ts = build $ fixedF 1 $ meanAvg ts
-    let fmtResult :: String -> [NominalDiffTime] -> IO ()
-        fmtResult title ts = fmtLn ("    "+|title|+" - "+|buildResult ts|+" ms")
+    fmtLn "Latencies for 2 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 2)
 
-    fmtLn "Latencies for two fixture wallets scenario"
+    fmtLn "Latencies for 10 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 10)
 
-    withUtf8Encoding $ benchWithServer logging $ \ctx -> do
-        (wal1, wal2) <- twoFixtureWallet ctx
+    fmtLn "Latencies for 100 fixture wallets scenario"
+    runScenario logging tvar (nFixtureWallet 100)
+
+    fmtLn "Latencies for 2 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 10)
+
+    fmtLn "Latencies for 2 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 20)
+
+    fmtLn "Latencies for 2 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 2 100)
+
+    fmtLn "Latencies for 10 fixture wallets with 10 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 10)
+
+    fmtLn "Latencies for 10 fixture wallets with 20 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 20)
+
+    fmtLn "Latencies for 10 fixture wallets with 100 txs scenario"
+    runScenario logging tvar (nFixtureWalletWithTxs 10 100)
+  where
+    nFixtureWallet n ctx = do
+        wal1 : wal2 : _ <- replicateM n (fixtureWallet ctx)
+        pure (wal1, wal2)
+
+    nFixtureWalletWithTxs n m ctx = do
```

Comment:
I think this needs a comment because it's not immediately obvious what the fixture setup function does.

Reply:
added corresponding comments